pubtools/pulplib/_impl/client/client.py

```diff
@@ -179,6 +181,36 @@ def search_repository(self, criteria=None):
             response_f, lambda data: self._handle_page(Repository, search, data)
         )
 
+    def _do_upload_file(self, upload_id, repo_id, filename):
+        tasks_f = f_return([])
+        offset = 0
+
+        def _do_next_part_upload(accumulated_tasks, upload_data):
+            data, upload_id, offset = upload_data
+            upload_task_f = self._do_upload(data, upload_id, offset)
```

Comment:
This doesn't look right since do_upload doesn't return tasks.  Does it?

Reply:
Right, it's a mistake here... 
it returns a request wrapped in a future, so I guess I actually don't need f_map here, as_completed should be enough for me?

Reply:
ah, I don't even need as_completed here, request_executor has done that for me.

Reply:
I think you don't actually care about the return value here, you just use the future to judge whether the operation completed.

In such cases I'm usually following the convention of "<verb>-ed",  e.g. "uploaded_f" is a future which is completed (or failed) when upload completes (or failed).